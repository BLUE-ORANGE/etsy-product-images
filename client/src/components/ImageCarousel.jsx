import React from 'react';
import ForwardButton from './ForwardButton';

const ImageCarousel = (props) => {
  let image = props.images.filter(img => img.focused === true);
  image = image[0].imageUrl;
  const componentClasses = ['carousel-image'];
  componentClasses.push('show');
  return (
    <div>
      <img
        alt="cat"
        style={{
          objectFit: 'scale-down',
          height: '456px',
          width: '570px',
          backgroundColor: '#f5f5f1',
          borderRadius: '3px',
        }}
        className={componentClasses.join(' ')}
        src={image}
      />
    </div>
  );
};


export default ImageCarousel;
