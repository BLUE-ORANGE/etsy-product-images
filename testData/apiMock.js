module.exports.testData = [
  {
    id: 108,
    imageUrl: 'https://i.imgur.com/zmTGRu1.jpg',
    focused: true,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  }, {
    id: 247,
    imageUrl: 'https://i.imgur.com/V1XNLjh.jpg',
    focused: false,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  }, {
    id: 731,
    imageUrl: 'https://i.imgur.com/jpc5JTA.jpg',
    focused: false,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  },
];
