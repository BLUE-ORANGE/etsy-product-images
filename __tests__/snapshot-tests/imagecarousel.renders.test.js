import React from 'react';
import renderer from 'react-test-renderer';
import ImageCarousel from '../../client/src/components/ImageCarousel';
import testData from '../../testData/apiMock';

it('ImageCarousel renders images', () => {
  const tree = renderer
    .create(<ImageCarousel images={testData.testData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
