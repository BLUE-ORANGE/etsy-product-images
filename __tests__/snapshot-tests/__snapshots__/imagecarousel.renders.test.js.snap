// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel renders images 1`] = `
<div>
  <img
    alt="cat"
    className="carousel-image show"
    src="https://i.imgur.com/zmTGRu1.jpg"
    style={
      Object {
        "backgroundColor": "#f5f5f1",
        "borderRadius": "3px",
        "height": "456px",
        "objectFit": "scale-down",
        "width": "570px",
      }
    }
  />
</div>
`;
